examples/proc_fieldSweep.py

```diff
@@ -1,142 +1,132 @@
+    print("s's dimlabels are ", s.dimlabels)
+    #assert set(s.getaxis("indirect").dtype.names) == {	
+    #    "Field",	
+    #    "carrierFreq",	
+    #}, "'indirect' axis should be a structured array that stores the carrier frequency and the field"	 
+    all_offsets = np.zeros(len(s.getaxis("indirect")))
+    carrier_freq_MHz = []
     for z in range(len(s.getaxis("indirect"))):
-        fl.plot(s["indirect", z], ax=ax_list[1])
-        if z == 0:
-            ax_list[1].axvline(color="k", x=freq_slice[0])
-            ax_list[1].axvline(color="k", x=freq_slice[-1])
-        offset = s["indirect", z].C.mean("nScans").argmax("t2").item()
+        fl.next("Field slicing")
+        if "nScans" in s.dimlabels:
+            fl.plot(s["indirect", z].C.mean("nScans"), label="scan %d" % z)
+            offset = s["indirect", z].C.mean("nScans").argmax("t2").item()
+        else:
+            fl.plot(s["indirect", z], label="scan %d" % z)
+            offset = s["indirect", z].C.argmax("t2").item()
         all_offsets[z] = offset
-        carrier_freq_MHz = s.getaxis("indirect")[z]["carrierFreq"]
-        field = s.getaxis("indirect")[z]["Field"]
-        nu_rf = carrier_freq_MHz - offset / 1e6  # because in order to make my
-        #                                       offset more negative, I
-        #                                       increase my field, or decrease
-        #                                       my carrier -- in other words
-        #                                       the frequency axis for our FID
-        #                                       is the negative of (ν_RF-ν_carrier)
-        nu_NMR.append(nu_rf)
-    ax_list[1].set_title("Field Slicing")
+        carrier_freq_MHz = s["indirect"][:][z]
     s.ift("t2")
-    s *= exp(
+    s *= np.exp(
         -1j
         * 2
-        * pi
-        * nddata(all_offsets, [-1], ["indirect"])
+        * np.pi
+        * psd.nddata(all_offsets, [-1], ["indirect"])
         * s.fromaxis("t2")
-    )
+    )    
     s.ft("t2")
-    # }}}
-    fl.plot(s.C.mean("nScans"), ax=ax_list[2])
-    s = s["t2":freq_slice].mean("nScans").integrate("t2")
-    # }}}
-    # {{{convert x axis to ppt = v_NMR/v_ESR
-    ppt = nu_NMR / nu_B12_GHz
-    s.setaxis("indirect", ppt)
-    s.set_units("indirect", "MHz/GHz")
-    s.rename("indirect", "resonance ratio")
-    # }}}
-    # {{{Fitting
-    fl.plot(s, "o-", ax=ax_list[3])
-    ax_list[3].set_title("Field Sweep ppt")
-    fitting = s.polyfit("resonance ratio", order=2)
-    field_fine = nddata(
-        r_[
-            s.getaxis("resonance ratio")[0] : s.getaxis("resonance ratio")[
-                -1
-            ] : 100j
-        ],
-        "resonance ratio",
-    )
-    polyline = fl.plot(
-        field_fine.eval_poly(fitting, "resonance ratio"),
-        label="fit",
-        ax=ax_list[3],
+    peak_range = nu_B12_GHz
+    peak_range = 1.2 * np.r_[-0.5, 0.5] * peak_range + np.mean(peak_range)
+    frq_slice = peak_range
```

Comment:
this needs to be changed to take the new peak_range (not this manually determined one)